docs_src/design/ucr/Microservice-Authentication.md

```diff
@@ -0,0 +1,145 @@
+
+  _PR.AC-1: Identities and credentials are issued, managed, verified,
+  revoked, and audited for authorized devices, users, and processes._
+
+
+### Target Users
+- Device Owner
+- Device User
+- Device Maintainer
+- Service Provider
+
+### Description
+
+Microservice authentication provides the following benefits,
+which are potentially valuable to all of the listed target users:
+
+- Provides a defense against malware running on the device,
+  as currently there is no mechanism to ensure that only
+  authorized users or processes are allowed to invoke EdgeX services.
+
+- Provides greater auditability as to who initiated a particular
+  action on the device.
+
+- Depending on implementation, may provide a way to revoke
+  access that was previously granted,
+  or allow customers to tie in to enterprise
+  identity management systems.
+
+For purposes of this UCR, microservice authentication implies
+that the receiving microservice has access to the identity
+of the caller and can write program logic based on that identity.
+
+
+### Existing solutions
+
+Microservice authentication is currently implemented around two primary vectors:
+
+- Token-based authentication schemes.
+  Initiator sends an identifier along with a request to the receiver.
+  The identifier is cryptographically validated using a key trusted by
+  the receiver, or the receiver asks a trusted third party to verify the identifier.
+
+  A benefit of token-based authentication schemes is identity delegation,
+  whereby the identifier can be passed through a chain of calls to
+  preserve the identity of the original initiator.
+  The identifier can often be tunneled through other protocols.
+  Another benefit of token-based authentication is that
+  it flows easily through a web application firewall.
+
+  A drawback of token-based authentication is that due to MITM threats,
+  token-based authentication over an unencrypted network is insecure.
+  Another drawback of token-based authentication is that it is unidirectional:
+  the receiver can authenticate the initiator, but not vice-versa.
+
+- End-to-end encryption schemes.
+  Both the initiator and the receiver participate in a session-oriented message exchange over an encrypted transport,
+  where both parties cryptographically validate each other's identity.
+  
+  A benefit of end-to-end encryption schemes is that it enforces proof-of-possession
+  (of a cryptographic key), thus preventing token-stealing attacks.
+
+  End-to-end encryption schemes are strongest when implemented at the application level,
+  as it does not require trust in the underlying network,
+  with the potential drawback that if network-level encryption is used,
+  or if all communication is done via local IPC interfaces,
+  the system may waste processing power on redundant encryption.
+  Additionally, authentication schemes based on end-to-end encryption
+  may complicate debugging because removing encryption also removes the authentication.
```

Comment:
I'd imagine in other zero trust overlays you cannot remove the encryption and continue to use the overlay. That's the case with OpenZiti. It __most definitely__ makes debugging more difficult though. You really can't inspect it anywhere except at the very beginning or very end of the communication channel. But, you can switch the application protocol OFF. I.e., you can go from https to plain http. Doing that does not break nor remove the authentication of a zero trust overlay, it does remove that "third layer" of encryption, but the traffic is still very safe as it traverses the zero trust overlay.  Sometimes you just cannot break/remove the secure transport or simply don't want to (because sometimes application layer encryption is 'the actual problem'). In those cases, it's definitely more difficult to diagnose and debug.

That's a long way for me to say, I don't think removing application-layer encryption would remove the authentication of the service, if you're using a zero trust overlay, like OpenZiti. (hope that makes sense)

Reply:
See next update.